src/main/java/com/endava/web/StudentController.java

```diff
@@ -0,0 +1,64 @@
+package com.endava.web;
+
+import com.endava.repository.Student;
+import com.endava.service.StudentService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.ResponseBody;
+import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.server.ResponseStatusException;
+
+import java.util.List;
+
+@RestController
+@RequestMapping("/student")
+public class StudentController {
+
+    private StudentService studentService;
+
+    @Autowired
+    public StudentController(StudentService studentService) {
+        this.studentService = studentService;
+    }
+
+    @RequestMapping(method = RequestMethod.GET, produces = "application/json")
+    public Student findStudentById(@RequestParam int id) {
+        if (studentService.findStudentById(id) == null) {
+            throw new ResponseStatusException(HttpStatus.NOT_FOUND, "Student with id=" + id + " doesn't exist.");
+        }
+        return studentService.findStudentById(id);
+    }
+
+    @RequestMapping(value = "/name", method = RequestMethod.GET, produces = "application/json")
```

Comment:
Same here, "name" should only be path param or query param, not both

Reply:
When /name is deleted I get this exception when starting then application:  
`
Invocation of init method failed; nested exception is java.lang.IllegalStateException: Ambiguous mapping. Cannot map 'studentController' method com.endava.web.StudentController#findStudentsByName(String) to {GET /student, produces [application/json]}: There is already 'studentController' bean method com.endava.web.StudentController#findStudentById(int) mapped.`

Reply:
That is completely expected. You have two mappings that are exactly the same, they need to be the same endpoint.
You should have one endpoint that takes two different params, if client provides ?id=111 you lookup by id, if it provides ?name=name you lookup by name. 
If it provides both ?name=name&id=id, there are several options here, think whats the best one and use that.

Second option will be to make them path params not query params, that way you can have two separate endpoints:
GET  example.net/user/id/{id}
GET example.net/user/name/{name}

There are few more options, like having different accept/produce types, but they are mostly workarounds.

Reply:
Thank you for suggestions. I have chosen 2nd solution because `findStudentById` returns String but `findStudentsByName` returns List.